assets/js/script.js

```diff
@@ -0,0 +1,178 @@
+// $(document).ready(function() {
+// })
+
+// Thing to consider is that the program uses 3 states -
+// 2 - not running
+// 1 - play mode
+// 0 - die mode - no input is accepted.
+
+
+$(function () {
+
+    // saving dom objects to variables
+  var $container = $('.container')
+  var $bird = $('.bird')
+  var pole = $('.pole')
+  var poleTop = $('#pole_1')
+  var poleBottom = $('#pole_2')
+  var score = $('#score')
+  var restartBtn = $('#restart-btn')
+  var $birdHeight = parseInt($bird.height())
+  var conWidth = parseInt($container.width())
+  var conHeight = parseInt($container.height())
+  var polePos = parseInt(pole.css('right'))
+  var pole_initial_height = parseInt(pole.css('height'))
+  var $birdLeft = parseInt($bird.css('left'))
+
+  var gameState = 2
+  var speed = 0
+  var scoreAdd = false
+  var stopKeyDown = false;
+
+  function start () {
+    birdFlap()
+  }
+
+  // audio
+  function spop() {
+    document.getElementsByClassName('startpop')[0].play()
+  }
+
+  function flapsound() {
+    // document.getElementsByClassName('birdFlapping')[0].currentTime = 1.3
+    document.getElementsByClassName('birdFlapping')[0].pause()
+    document.getElementsByClassName('birdFlapping')[0].play()
+  }
+
+  function playBgSound() {
+    document.getElementsByClassName('play_bgsound')[0].volume = 0.8
+    document.getElementsByClassName('play_bgsound')[0].play()
+  }
+
+  function hitOver() {
+    document.getElementsByClassName('hit')[0].play()
+  }
+  // audio
+
+  spop()
+
+  // start instruction popup
+  $startBtn = $('.start')
+  $startBtn.on('click', () => {
+    document.getElementsByClassName('startpop')[0].pause()
+    stopKeyDown = true;
+    $('.start_pop').hide()
+    playBgSound ()
+    speed = 20
+  })
+
+	$container.on('mousedown', function () {
+    start()
+    birdFlap ()
+    flapsound()
+	})
+
+	$(this).on('keydown', function (e) {
+    if (stopKeyDown) {
+      start()
+      birdFlap ()
+      flapsound()
+      if (e.keyCode === 32) {
+        if (gameState > 1) return
+        if (gameState === 2) gameState = 1
+      }
+    }
+  })
+
+
+  var polePosInterval = setInterval(poleMove, 50)
+
+  var birdPosInterval = setInterval(function () {
```

Comment:
What will happen if I do...? and why?

`var birdPosInterval = setInterval(collisionDetection, 50)`